src/logger.ts

```diff
@@ -74,9 +74,22 @@ type StrictLogger = Omit<winston.Logger, keyof typeof LOG_LEVELS> & {
   [K in keyof typeof LOG_LEVELS]: StrictLogMethod;
 };
 
+/**
+ * Extracts the actual message string from potentially nested info objects
+ */
+function extractMessage(info: any): string {
```

Comment:
Why `any` instead of something like `info: { message: { message?: string } | null }`?

Reply:
I implemented 

```ts 
type LogInfo = {
  message: string | { message?: any } | null | any;
  [key: string]: any;
};
```

but I think type safety is good but the any type is actually reasonable here given Winston's flexible log object structure.